src/m365/entra/commands/user/user-registrationdetails-list.ts

```diff
@@ -0,0 +1,327 @@
+    }
+
+    if (options.isSelfServicePasswordResetCapable !== undefined) {
+      filters.push(`isSelfServicePasswordResetCapable eq ${options.isSelfServicePasswordResetCapable ? true : false}`);
+    }
+
+    if (options.isSelfServicePasswordResetEnabled !== undefined) {
+      filters.push(`isSelfServicePasswordResetEnabled eq ${options.isSelfServicePasswordResetEnabled ? true : false}`);
+    }
+
+    if (options.isSelfServicePasswordResetRegistered !== undefined) {
+      filters.push(`isSelfServicePasswordResetRegistered eq ${options.isSelfServicePasswordResetRegistered ? true : false}`);
+    }
+
+    if (options.isSystemPreferredAuthenticationMethodEnabled !== undefined) {
+      filters.push(`isSystemPreferredAuthenticationMethodEnabled eq ${options.isSystemPreferredAuthenticationMethodEnabled ? true : false}`);
+    }
+
+    const methodsRegistered: string[] = [];
+    if (options.methodsRegistered) {
+      const methods = options.methodsRegistered.split(',').map(m => m.trim());
+      methods.forEach(method => {
+        methodsRegistered.push(`methodsRegistered/any(m:m eq '${method}')`);
+      });
+    }
+
+    if (methodsRegistered.length > 0) {
+      filters.push(`(${methodsRegistered.join(' or ')})`);
+    }
+
+    const systemPreferredAuthenticationMethods: string[] = [];
+    if (options.systemPreferredAuthenticationMethods) {
+      const methods = options.systemPreferredAuthenticationMethods.split(',').map(m => m.trim());
+      methods.forEach(method => {
+        systemPreferredAuthenticationMethods.push(`systemPreferredAuthenticationMethods/any(m:m eq '${method}')`);
+      });
+    }
+
+    if (systemPreferredAuthenticationMethods.length > 0) {
+      filters.push(`(${systemPreferredAuthenticationMethods.join(' or ')})`);
+    }
+
+    const userUPNs: string[] = [];
+    if (userUpns.length > 0) {
+      userUpns.forEach(upn => {
+        userUPNs.push(`userPrincipalName eq '${formatting.encodeQueryParameter(upn)}'`);
+      });
+    }
+
+    if (options.userPrincipalNames) {
+      const upns = options.userPrincipalNames.split(',').map(m => m.trim());
+      upns.forEach(upn => {
+        userUPNs.push(`userPrincipalName eq '${formatting.encodeQueryParameter(upn)}'`);
+      });
+    }
+
+    if (userUPNs.length > 0) {
+      filters.push(`(${userUPNs.join(' or ')})`);
+    }
+
+    const userPreferredMethodForSecondaryAuthentication: string[] = [];
+    if (options.userPreferredMethodForSecondaryAuthentication) {
+      const methods = options.userPreferredMethodForSecondaryAuthentication.split(',').map(m => m.trim());
+      methods.forEach(method => {
+        userPreferredMethodForSecondaryAuthentication.push(`userPreferredMethodForSecondaryAuthentication eq '${method}'`);
+      });
+    }
+
+    if (userPreferredMethodForSecondaryAuthentication.length > 0) {
+      filters.push(`(${userPreferredMethodForSecondaryAuthentication.join(' or ')})`);
+    }
```

Comment:
This can be simplified


```suggestion
    const userPreferredMethodForSecondaryAuthentication = options.userPreferredMethodForSecondaryAuthentication?
      .split(',').map(method => `userPreferredMethodForSecondaryAuthentication eq ${method.trim()}`);  
    const userPreferredMethodForSecondaryAuthenticationFilter = userPreferredMethodForSecondaryAuthentication?.join(' or ');  

    if (userPreferredMethodForSecondaryAuthenticationFilter) {  
      filters.push(`(${userPreferredMethodForSecondaryAuthenticationFilter})`);  
    }  
```